### ISSUE TITLE
[Bug] `validate` should check passed in model type

### ISSUE THREAD

**Issue by Pirulax** [sanic-org/sanic-ext#257]
**Describe the bug**
The `validate` function should check the type of the passed in schema.
In my case doing this:
```py
import sanic as s
from sanic_ext import validate

class MyRESTMethodArgs:
    id : str
@bp.post('/create')
@validate(json=MyRESTMethodArgs)
async def post_create(req : s.Request, body : MyRESTMethodArgs):
    pass # Business logic
```
Fails miserably with:
```log
File "<irrelevant>\.venv\Lib\site-packages\sanic_ext\extras\validation\decorator.py", line 73, in decorated_function
    retval = f(*args, **kwargs)
```
Inspecting it closer, it turns out `schemas` looks like this:
```py
{'json': {}, 'form': None, 'query': None}
```
I assume the issue is with `generate_schema` for some reason just returning `{}` instead of raising an exception

**To Reproduce**
See above

**Expected behavior**
`validate` should fail if it encounters a schema model it can't process (As according to the docs, a simple Python class can't be used)

**Environment (please complete the following information):**
mode: debug, single worker
server: sanic, HTTP/1.1
python: 3.12.4
platform: Windows-11-10.0.22631-SP0 
packages: sanic-routing==23.12.0, sanic-ext==23.12.0

**Comment by goodki-d** [comment]
Yeah this behaviour checks out looking at internal `make_schema` function:

https://github.com/sanic-org/sanic-ext/blob/fd85ada01b71a2f1318794eea7c252369caf54be/sanic_ext/extras/validation/schema.py#L41-L69

That is, passing a bare python class will not raise error but return the empty dict.

**Comment by goodki-d** [comment]
> `validate` should fail if it encounters a schema model it can't process (As according to the docs, a simple Python class can't be used)

I think only _pydantic, msgspec, attrs and dataclasses_ are supported

We can add a check here
https://github.com/sanic-org/sanic-ext/blob/fd85ada01b71a2f1318794eea7c252369caf54be/sanic_ext/extras/validation/setup.py#L48-L55

maybe like
```diff
+ if isclass(param) and not is_attrs(param) and not is_dataclass(param):
+       raise SchemaNotSupported

return make_schema({}, param) if isclass(param) else param
